plugins/woocommerce-blocks/tests/e2e/tests/price-filter/price-filter.block_theme.spec.ts

```diff
@@ -0,0 +1,409 @@
+		{ page, admin, editor, templateApiUtils, editorUtils },
+		use
+	) => {
+		const pageObject = new ProductCollectionPage( {
+			page,
+			admin,
+			editor,
+			templateApiUtils,
+			editorUtils,
+		} );
+		await use( pageObject );
+	},
+} );
+
+test.describe( `${ blockData.name } Block - editor side`, () => {
+	test.beforeEach( async ( { admin, editor } ) => {
+		await admin.createNewPost();
+		await editor.insertBlock( {
+			name: 'woocommerce/filter-wrapper',
+			attributes: {
+				filterType: 'price-filter',
+				heading: 'Filter By Price',
+			},
+		} );
+		await editor.openDocumentSettingsSidebar();
+	} );
+
+	test( "should allow changing the block's title", async ( { page } ) => {
+		const textSelector =
+			'.wp-block-woocommerce-filter-wrapper .wp-block-heading';
+
+		const title = 'New Title';
+
+		await page.locator( textSelector ).fill( title );
+
+		await expect( page.locator( textSelector ) ).toHaveText( title );
+	} );
+
+	test( 'should allow changing the display style', async ( {
+		page,
+		editorUtils,
+		editor,
+	} ) => {
+		const priceFilterControls = await editorUtils.getBlockByName(
+			'woocommerce/price-filter'
+		);
+		await editor.selectBlocks( priceFilterControls );
+
+		await expect(
+			priceFilterControls.getByRole( 'textbox', {
+				name: 'Filter products by minimum',
+			} )
+		).toBeVisible();
+
+		await expect(
+			priceFilterControls.getByRole( 'textbox', {
+				name: 'Filter products by maximum',
+			} )
+		).toBeVisible();
+
+		await page
+			.getByLabel( 'Price Range Selector' )
+			.getByText( 'Text' )
+			.click();
+
+		await expect(
+			priceFilterControls.getByRole( 'textbox', {
+				name: 'Filter products by minimum',
+			} )
+		).toBeHidden();
+
+		await expect(
+			priceFilterControls.getByRole( 'textbox', {
+				name: 'Filter products by maximum',
+			} )
+		).toBeHidden();
+	} );
+
+	test( 'should allow toggling the visibility of the filter button', async ( {
+		page,
+		editorUtils,
+		editor,
+	} ) => {
+		const priceFilterControls = await editorUtils.getBlockByName(
+			blockData.slug
+		);
+		await editor.selectBlocks( priceFilterControls );
+
+		await expect(
+			priceFilterControls.getByRole( 'button', {
+				name: 'Apply',
+			} )
+		).toBeHidden();
+
+		await page.getByText( "Show 'Apply filters' button" ).click();
+
+		await expect(
+			priceFilterControls.getByRole( 'button', {
+				name: 'Apply',
+			} )
+		).toBeVisible();
+	} );
+} );
+
+test.describe( `${ blockData.name } Block - with All products Block`, () => {
+	test.beforeEach( async ( { admin, page, editor } ) => {
+		await admin.createNewPost( { legacyCanvas: true } );
+		await editor.insertBlock( { name: 'woocommerce/all-products' } );
+		await editor.insertBlock( {
+			name: 'woocommerce/filter-wrapper',
+			attributes: {
+				filterType: 'price-filter',
+				heading: 'Filter By Price',
+			},
+		} );
+		await editor.publishPost();
+		const url = new URL( page.url() );
+		const postId = url.searchParams.get( 'post' );
```

Comment:
`editor.publishPost()` returns the post ID so we can make a oneliner here:
```suggestion
		const postId = await editor.publishPost();
```